frontend/packages/dev-console/src/components/import/pipeline/pipeline-template-utils.ts

```diff
@@ -24,28 +29,34 @@ export const createPipelineForImportFlow = async (formData: GitImportFormData) =
     git,
     pipeline,
   } = formData;
-  const template = _.cloneDeep(pipeline.template);
+  const template = _.cloneDeep(pipeline.template) as Pipeline;
 
   template.metadata = {
     name: `${name}`,
     namespace,
     labels: { ...template.metadata.labels, 'app.kubernetes.io/instance': name },
   };
 
-  template.spec.params =
-    template.spec.params &&
-    template.spec.params.map((param) => {
-      if (param.name === 'APP_NAME') {
-        param.default = name;
-      }
-      return param;
-    });
-
-  try {
+  template.spec.params = template.spec.params?.map((param) => {
+    switch (param.name) {
+      case 'APP_NAME':
+        return { ...param, default: name };
+      case 'GIT_REPO':
+        return { ...param, default: git.url };
+      case 'GIT_REVISION':
+        return { ...param, default: git.ref || 'master' };
+      case 'IMAGE_NAME':
+        return { ...param, default: getImageUrl(name, namespace) };
+      default:
+        return param;
+    }
+  });
+
+  if (template.spec?.resources?.find((r) => r.type === 'git' && r.name === 'app-source')) {
     await createGitResource(git.url, namespace, git.ref);
+  }
+  if (template.spec?.resources?.find((r) => r.type === 'image' && r.name === 'app-image')) {
     await createImageResource(name, namespace);
-  } catch (err) {
-    throw err;
```

Comment:
Hah, a fair catch. Not really doing anything if we are just catching and rethrowing. Good cleanup :)